### DISCUSSION TITLE
Create Share using Flow

### DISCUSSION THREAD

**Discussion by rickysullivan** [directus/directus#17480] [Archive]
Say I'm creating a new item in a collection, is there a way to use Flows to programmatically create a Share for that item?

**Comment by rijkvanzanten** [comment]
Great question. I _think_ you should be able to create a regular item from the "create item" operation, and use `directus_shares` as the target 🤔

**Reply by rickysullivan** [reply]
I’ll give it a crack.

**Reply by rickysullivan** [reply]
How can I specify a UUID for the row? The id column for directus_shares doesn't have a default value.

**Reply by rickysullivan** [reply]
I managed to do this using a script block which imports the UUID package.

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/437480/218331264-233f4e8d-f6c0-4767-8423-2e3740b2a61f.png">

```js
const { v4: uuidv4 } = require('uuid');
module.exports = async function(data) {
 	const uuid = uuidv4();
	return { id: uuid };
}
```

Kinda feels like `$UUID` should be a special variable that you can use for Flows.

**Reply by rickysullivan** [reply]
Oh and to target `directus_shares` model, I had to make the edit in the DB. The Collection dropdown in Directus doesn't list system models.

**Reply by rijkvanzanten** [reply]
If you don't provide an ID, it should be auto-generated for collections using a UUID primary key 🤔
